--- a/languages/cpp/parser/rpp/CMakeLists.txt
+++ b/languages/cpp/parser/rpp/CMakeLists.txt
@@ -22,6 +22,7 @@ set(kdevcpprpp_LIB_SRCS
 #       to the plugin target
 kde4_add_library(kdev4cpprpp SHARED ${kdevcpprpp_LIB_SRCS})
 target_link_libraries(kdev4cpprpp ${KDE4_KDECORE_LIBS} ${KDEVPLATFORM_LANGUAGE_LIBRARIES} ${KDE4_KTEXTEDITOR_LIBS} )
+set_target_properties(kdev4cpprpp PROPERTIES SOVERSION ${PRIVATE_LIBS_SOVERSION})
 install(TARGETS kdev4cpprpp ${INSTALL_TARGETS_DEFAULT_ARGS} )
 
 ########### install files ###############
--- a/languages/cpp/parser/CMakeLists.txt
+++ b/languages/cpp/parser/CMakeLists.txt
@@ -35,4 +35,5 @@ set(kdevcppparser_LIB_SRCS
 #       to the plugin target
 kde4_add_library(kdev4cppparser SHARED ${kdevcppparser_LIB_SRCS})
 target_link_libraries(kdev4cppparser ${KDEVPLATFORM_LANGUAGE_LIBRARIES} ${KDE4_KDECORE_LIBS} ${KDE4_KTEXTEDITOR_LIBS} kdev4cpprpp )
+set_target_properties(kdev4cppparser PROPERTIES SOVERSION ${PRIVATE_LIBS_SOVERSION})
 install(TARGETS kdev4cppparser ${INSTALL_TARGETS_DEFAULT_ARGS} )
--- a/languages/cpp/cppduchain/CMakeLists.txt
+++ b/languages/cpp/cppduchain/CMakeLists.txt
@@ -62,6 +62,7 @@ target_link_libraries(kdev4cppduchain 
 		kdev4cppparser 
 		${KDE4_KTEXTEDITOR_LIBS} 
 		kdev4cpprpp)
+set_target_properties(kdev4cppduchain PROPERTIES SOVERSION ${PRIVATE_LIBS_SOVERSION})
 install(TARGETS kdev4cppduchain ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
--- a/projectmanagers/cmake/CMakeLists.txt
+++ b/projectmanagers/cmake/CMakeLists.txt
@@ -71,6 +71,7 @@ target_link_libraries( kdev4cmakecommon 
                         ${KDE4_KTEXTEDITOR_LIBS} ${KDEVPLATFORM_INTERFACES_LIBRARIES} ${KDEVPLATFORM_PROJECT_LIBRARIES}
                         ${QT_QTSCRIPT_LIBRARY} ${KDEVPLATFORM_LANGUAGE_LIBRARIES}
                         )
+set_target_properties(kdev4cmakecommon PROPERTIES SOVERSION ${PRIVATE_LIBS_SOVERSION})
 
 kde4_add_ui_files( cmakemanager_SRCS ${cmakemanager_UI} )
 kde4_add_plugin( kdevcmakemanager ${cmakemanager_SRCS} )
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,7 @@ set(CMAKE_MODULE_PATH ${KDevelop_SOURCE_
 set( KDEVELOP_VERSION_MAJOR 3 )
 set( KDEVELOP_VERSION_MINOR 9 )
 set( KDEVELOP_VERSION_PATCH 97 )
+set(PRIVATE_LIBS_SOVERSION ${KDEVELOP_VERSION_MAJOR}.${KDEVELOP_VERSION_MINOR}.${KDEVELOP_VERSION_PATCH})
 
 set(KDE_MIN_VERSION "4.3.0")
 find_package(KDE4 4.3.0 REQUIRED)
